cleaner/docker.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2022 Red Hat, Inc. and/or its affiliates.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cleaner
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"time"
+)
+
+// https://docs.docker.com/engine/api/latest/
+func (d Docker) GetClient() (*client.Client, error) {
+	return client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+}
+
+func (d Docker) GetClientRemoteFromEnv(host string) (*client.Client, error) {
+	return client.NewClientWithOpts(client.FromEnv, client.WithHost(host))
+}
+
+func (d Docker) GetClientRemote(host string, cacertPath string, certPath string, keyPath string) (*client.Client, error) {
+	return client.NewClientWithOpts(client.WithHost(host), client.WithAPIVersionNegotiation(), client.WithTLSClientConfig(cacertPath, certPath, keyPath))
+}
+
+func (d Docker) GetAllImages() []types.ImageSummary {
+	images, err := d.Connection.ImageList(context.Background(), types.ImageListOptions{All: true})
+	if err != nil {
+		logrus.Errorf("%s \n", err)
+	}
+	return images
+}
+
+func (d Docker) GetImagesFiltered(args filters.Args) ([]types.ImageSummary, error) {
+	images, err := d.Connection.ImageList(context.Background(), types.ImageListOptions{Filters: args})
+	if err != nil {
+		logrus.Errorf("%s \n", err)
+		return nil, err
+	}
+	return images, nil
+}
+
+// Remove the images with tag <none>:<none>
+func (d Docker) RemoveImagesUntagged() {
+	images := d.GetAllImages()
+	for _, image := range images {
+		if image.RepoTags != nil && image.RepoTags[0] == "<none>:<none>" && image.RepoDigests[0] == "<none>@<none>" {
+			item, err := d.Connection.ImageRemove(context.TODO(), image.ID, types.ImageRemoveOptions{PruneChildren: true, Force: true})
+			if err != nil {
+				logrus.Errorf("%s %s\n", err, item)
```

Comment:
see my comment at the interface declaration.

Reply:
fixed